embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/CurrentSessionSpanImpl.kt

```diff
@@ -36,8 +36,12 @@ internal class CurrentSessionSpanImpl(
     private val sessionSpan: AtomicReference<PersistableEmbraceSpan?> = AtomicReference(null)
 
     override fun initializeService(sdkInitStartTimeMs: Long) {
-        synchronized(sessionSpan) {
-            sessionSpan.set(startSessionSpan(sdkInitStartTimeMs))
+        if (sessionSpan.get() == null) {
+            synchronized(sessionSpan) {
+                if (sessionSpan.get() == null) {
```

Comment:
Is it worth formalising this pattern in a separate PR? It's raised a couple of questions in the past - I understand the rationale, but others might not

```
private inline fun doubleCheckLock(lock: Any, predicate: () -> Boolean, action: () -> Unit) {
        if (predicate()) {
            return
        }
        synchronized(lock) {
            if (predicate()) {
                return
            }
            action()
        }
    }
```

Reply:
Yeah we should! We'll have to be careful how we implement these - we don't want a dangling reference in an `AtomicReference` causing leaks.